File: src/Pascal's Triangle/Pascal's Triangle.py
class Solution:
	# @return a list of lists of integers
	def generate(self, numRows):
		ret = [[1] for i in range(numRows) ]
		if numRows == 1: return ret 

		for i in range(1, numRows):
			for j in range(1, i+1):
				if j == i: ret[i].append(1)
				else: ret[i].append( ret[i-1][j-1] + ret[i-1][j] )
		return ret 

File: src/Pascal's Triangle/Pascal's Triangle.cpp
class Solution {
	public:
		vector<vector<int> > generate(int numRows) {
			vector< vector<int> > ret ( numRows, vector<int>(1,1) );
			if(numRows == 1) return ret;

			for (int i = 1; i < ret.size(); i++) 
			{
				for(int j = 1 ; j <= i ; j++ )
				{
					if(j == i) ret[i].push_back(1);
					else ret[i].push_back( ret[i-1][j-1] + ret[i-1][j] );
				}
			}
			return ret;   
		}
};	
